dev/react-plugin-tools/bootstrap.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Bootstrap React Plugin CLI Tool.
+
+This script provides a command-line interface to create new React UI plugin
+directories based on the airflow-core/ui project structure. It sets up all the
+necessary configuration files, dependencies, and basic structure for development
+with the same tooling as used in Airflow's core UI.
+"""
+
+from __future__ import annotations
+
+import argparse
+import re
+import shutil
+import sys
+from pathlib import Path
+
+
+def get_template_dir() -> Path:
+    """Get the template directory path."""
+    script_dir = Path(__file__).parent
+    template_dir = script_dir / "react_plugin_template"
+
+    if not template_dir.exists():
+        print(f"Error: Template directory not found at {template_dir}")
+        sys.exit(1)
+
+    return template_dir
+
+
+def replace_template_variables(content: str, project_name: str) -> str:
+    """Replace template variables in file content."""
+    return content.replace("{{PROJECT_NAME}}", project_name)
+
+
+def remove_apache_license_header(content: str, file_extension: str) -> str:
+    """Remove Apache license header from file content based on file type."""
+    if file_extension in [".ts", ".tsx", ".js", ".jsx"]:
```

Comment:
Haha, first this that pre-commit did was adding all the headers when using in the airflow repo :-)